### PR TITLE
[Merged by Bors] - feat: port CategoryTheory.Functor.EpiMono

### PR DESCRIPTION
**Pull request by mattrobball**
---

- [x] depends on: #2327
- [x] depends on: #2329

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by semorrison** [comment]
This PR/issue depends on:
* leanprover-community/mathlib4#2327
* leanprover-community/mathlib4#2329
By **[Dependent Issues](https://github.com/z0al/dependent-issues)** (🤖). Happy coding!

**Comment by semorrison** [comment]
This PR/issue depends on:
* ~~leanprover-community/mathlib4#2327~~
* ~~leanprover-community/mathlib4#2329~~
By **[Dependent Issues](https://github.com/z0al/dependent-issues)** (🤖). Happy coding!

File: scripts/start_port-macos.sh [hunk 1]
```diff
@@ -0,0 +1,86 @@
+#!/usr/bin/env bash
```
Comment (jcommelin) [inline scripts/start_port-macos.sh:1]: This shouldn't be checked in :smiley:

Comment (mattrobball) [reply scripts/start_port-macos.sh]: Oopa

**Review by jcommelin — APPROVED** [review]
Thanks :tada:

bors merge

**Pull request closed without merging**